core/net/rpl/rpl-ext-header.c

```diff
@@ -370,6 +374,12 @@ insert_srh_header()
   /* For simplicity, we use cmpri = cmpre */
   cmpri = 15;
   cmpre = 15;
+
+  if(node != NULL && node == root_node) {
```

Comment:
very minor comment: we already know that `root_node` is not null so I would only check `node == root_node`.